Issue96 Fix

Kurios opened the pull request:
#96

Akjosch commented:
Shouldn't the first check be for `botForces` being `null`? After all, that would cause an error down the line, while `specMissionEnemies` being `null` is already checked for and handled properly.

Kurios commented:
So the problem wasnt botForces bieng null.

The actual error was 

    Exception in thread "AWT-EventQueue-0" java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
    ...
    at mekhq.campaign.mission.AtBScenario.refresh(AtBScenario.java:719)

Fixed this by changing the order that we checked in the if by switching `if(botForces.get(0) != null && specMissionEnemies != null && specMissionEnemies.get(weight) != null)` to `if(specMissionEnemies != null && botForces.get(0) != null &&  specMissionEnemies.get(weight) != null)' so we do the check to see if we have even bothered to set enemies first.

In addition to that, I added code to have it generate enemies on the refresh, to help prevent that from being a problem. (The refresh function should have been called on the day advance, but I'm not sure why it wasn't. May have something to do with that special mission. Outside of that, if you didn't figure out how to assign people to the mission until day of, it made it hard to start the mission without having a way to force the force generation. Putting it in the gui refresh was the easiest solution. It only runs once, and doesn't regenerate  forces every time you click on the gui element. )
